modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java

```diff
@@ -220,6 +222,22 @@ else if (name.equals(AccountRoleConstants.ROLE_NAME_SUPPLIER)) {
 					new String[] {ActionKeys.ACCESS_IN_CONTROL_PANEL});
 			}
 
+			companyResourceActionIds.put(
+				"com.liferay.commerce.currency",
+				new String[] {
+					CommerceCurrencyActionKeys.MANAGE_COMMERCE_CURRENCIES
```

Comment:
Probably better to use the local service where it's needed instead of giving the supplier the ability to actually change currencies.
Even better would be having a VIEW permission on currencies, set as guest-default.
Align with @dsitu that faced a similar issue and used the local service

Reply:
hi @luca-pellizzon , I will identify and adjust accordingly.

Reply:
@luca-pellizzon , see commit e283358.